### DISCUSSION TITLE
Bootloader serial transfer of kernel.hex from the console for Windows

### DISCUSSION THREAD

**Discussion by ArnoldUK** [rsta2/circle#361] [Ideas]
I've written a small batch file for Windows which transfers the kernel generated hex file when using `make flash` for bootloader kernels. I know there is a python script to do this automatically but sometimes it is also useful to be able to do this from Windows. Also saves all the hassle of installing python and the missing serial scripts that don't get installed!. These commands can be typed directly into a console window (replace COM4 with the serial port assigned for the usb cable).
```
arm-none-eabi-objcopy kernel.elf -O ihex kernel.hex
mode COM4 /STATUS
mode COM4: BAUD=115200 PARITY=n DATA=8
rem set /p x="magicstring" <nul >\\.\COM4
set /p x="R" <nul >\\.\COM4
copy /A kernel.hex COM4
echo g > \\.\COM4
```
I've REMed out the "magicstring" line as I don't know how it works exactly but the `reboottool.py` script uses it to reboot the kernel. There is no progress shown for the file copy part so give it time to finish. I hope this contribution proves useful to someone and please feel free to add it to the repo.

**Comment by rsta2** [comment]
Thanks for providing this method to load the kernel image using the serial bootloader on Windows! There is another comfortable bootloader tool (Flashy), which runs on Windows, already integrated into Circle. It is described [here](https://github.com/rsta2/circle/blob/master/doc/bootloader.txt#L59).

**Comment by ArnoldUK** [comment]
Thank you. I see the alternatives all require installation of extra files and packages for Windows and paths to be setup correctly. I posted this method as it works without all that hassle even though it's a quick and dirty method so to speak.

I use Windows mostly and a Linux box dedicated for compile and builds. After an hour of frustration downloading and installing the python3 package (over 5K files BTW) I was still unable to run the python scripts as the serial library was not part of the install. I then spent another hour installing and setting up "pip" in order to install the serial library.

So, eventually I gave up and wrote this short batch file in 10 minutes.

**Reply by rsta2** [reply]
I understand. Good to have an easy-to-install option for the bootloader for Windows. Thanks again!
